llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -269,13 +269,15 @@ static bool isMergePassthruOpcode(unsigned Opc) {
 
 // Returns true if inactive lanes are known to be zeroed by construction.
 static bool isZeroingInactiveLanes(SDValue Op) {
+  // Skip bitcasts nodes
+  while (Op->getOpcode() == ISD::BITCAST)
+    Op = Op->getOperand(0);
+    
```

Comment:
Is this necessary?  The function is specific to predicate vectors so I'm not sure what bit casts are likely to be present.

Reply:
Yes I was wondering about that as well. I added it because `isConstantSplatVectorAllOnes` was doing it and wanted to preserve it. I am not sure whether we have any casts from other types to predicate type.

Reply:
I will remove that then